addons/stock_picking_batch/models/stock_picking_batch.py

```diff
@@ -95,6 +102,21 @@ def _compute_state(self):
             elif all(picking.state in ['cancel', 'done'] for picking in batch.picking_ids):
                 batch.state = 'done'
 
+    @api.depends('picking_ids.scheduled_date')
+    def _compute_scheduled_date(self):
+        """ scheduled_date = the earliest scheduled_date across all pickings. """
+        for batch in self:
+            dates_list = batch.picking_ids.filtered(lambda p: p.scheduled_date).mapped('scheduled_date')
+            if dates_list:
+                batch.scheduled_date = fields.Datetime.to_string(min(dates_list))
+            else:
+                batch.scheduled_date = False
+
+    @api.onchange('scheduled_date')
+    def onchange_scheduled_date(self):
```

Comment:
I'm following the same logic as was implemented in this: https://github.com/odoo/odoo/pull/52693/commits/49cefa79829695b0d7566707ca01e0cbbdfc6449 Where `onchange` is overwritten in `purchase.py` to prevent an update to one of the picking's scheduled dates from updating all of the batch picking's scheduled dates. It's confusing logic in my opinion:
- manually changing batch's scheduled date => automatically update all picking's scheduled dates
- manually update one of the picking's scheduled dates => automatically update the batch's scheduled date, but not the other pickings' scheduled dates